src/repr/src/adt/mz_acl_item.rs

```diff
@@ -0,0 +1,347 @@
+            write!(f, "{UPDATE_CHAR}")?;
+        }
+        if self.contains(AclMode::DELETE) {
+            write!(f, "{DELETE_CHAR}")?;
+        }
+        if self.contains(AclMode::USAGE) {
+            write!(f, "{USAGE_CHAR}")?;
+        }
+        if self.contains(AclMode::CREATE) {
+            write!(f, "{CREATE_CHAR}")?;
+        }
+        Ok(())
+    }
+}
+
+impl RustType<ProtoAclMode> for AclMode {
+    fn into_proto(&self) -> ProtoAclMode {
+        ProtoAclMode {
+            acl_mode: self.bits,
+        }
+    }
+
+    fn from_proto(proto: ProtoAclMode) -> Result<Self, TryFromProtoError> {
+        Ok(AclMode {
+            bits: proto.acl_mode,
+        })
+    }
+}
+impl Columnation for AclMode {
+    type InnerRegion = CloneRegion<AclMode>;
+}
+
+/// A list of privileges granted to a role in Materialize.
+///
+/// This is modelled after the AclItem type in PostgreSQL, but the OIDs are replaced with RoleIds
+/// because we don't use OID as persistent identifiers for roles.
+///
+/// See: <https://github.com/postgres/postgres/blob/7f5b19817eaf38e70ad1153db4e644ee9456853e/src/include/utils/acl.h#L48-L59>
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Hash, Deserialize)]
+pub struct MzAclItem {
+    /// Role that this item grants privileges to.
+    pub grantee: RoleId,
+    /// Grantor of privileges.
+    pub grantor: RoleId,
+    /// Privileges bit flag.
+    pub acl_mode: AclMode,
+}
+
+impl MzAclItem {
+    pub fn encode_binary(&self) -> Vec<u8> {
+        let mut res = Vec::new();
+        res.extend_from_slice(&self.grantee.encode_binary());
+        res.extend_from_slice(&self.grantor.encode_binary());
+        res.extend_from_slice(&self.acl_mode.bits().to_le_bytes());
+        res
+    }
+
+    pub fn decode_binary(raw: &[u8]) -> Result<MzAclItem, Error> {
+        if raw.len() != MzAclItem::binary_size() {
+            return Err(anyhow!(
+                "invalid binary size, expecting {}, found {}",
+                MzAclItem::binary_size(),
+                raw.len()
+            ));
+        }
+
+        let role_id_size = RoleId::binary_size();
+
+        let grantee = RoleId::decode_binary(&raw[0..role_id_size])?;
+        let raw = &raw[role_id_size..];
+        let grantor = RoleId::decode_binary(&raw[0..role_id_size])?;
+        let raw = &raw[role_id_size..];
+        let acl_mode = u64::from_le_bytes(raw.try_into()?);
+
+        Ok(MzAclItem {
+            grantee,
+            grantor,
+            acl_mode: AclMode { bits: acl_mode },
+        })
+    }
+
+    pub const fn binary_size() -> usize {
+        RoleId::binary_size() + RoleId::binary_size() + size_of::<AclMode>()
+    }
+}
+
+impl FromStr for MzAclItem {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let re: Regex = Regex::new(&format!(
+            r"(.*)=([{INSERT_CHAR},{SELECT_CHAR},{UPDATE_CHAR},{DELETE_CHAR},{USAGE_CHAR},{CREATE_CHAR}]*)/(.+)"
```

Comment:
```suggestion
            r"(.*)=(.*?)/(.+)"
```

Light preference for this because it makes this more obvious to understand mentally what's going on, and doesn't seem to be wrong because the parse fn below is going to assert the correct characters.

Would actually prefer a completely non-regex thing here instead. Could do simple string splitting on `=` then `/`, and let the lower parse fns handle validation of each part.

Reply:
I've updated it to not use regex. I was already on the fence about whether to use regex, so if you have a preference not to then I think it's good to switch.